src/main/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoImpl.java

```diff
@@ -0,0 +1,112 @@
+package org.pesmypetcare.webservice.dao;
+
+import com.google.api.services.calendar.model.Calendar;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+import org.pesmypetcare.webservice.error.CalendarAccessException;
+import org.pesmypetcare.webservice.firebaseservice.CalendarServiceFactory;
+import org.springframework.stereotype.Repository;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Marc Simó
+ */
+@Repository
+public class GoogleCalendarDaoImpl implements GoogleCalendarDao {
+    private com.google.api.services.calendar.Calendar service;
+    private CalendarServiceFactory factory;
+
+    public GoogleCalendarDaoImpl() {
+        factory = CalendarServiceFactory.getInstance();
+    }
+
+
+    @Override
+    public String createSecondaryCalendar(String accessToken, Calendar calendar) throws CalendarAccessException {
+        service = factory.initializeService(accessToken);
+        Calendar createdCalendar;
+        try {
+            createdCalendar = service.calendars().insert(calendar).execute();
+        } catch (IOException e) {
+            throw new CalendarAccessException("Error inserting secondary calendar", e.getMessage());
+        }
+        return createdCalendar.getId();
+    }
+
+    @Override
+    public void deleteSecondaryCalendar(String accessToken, String calendarId) throws CalendarAccessException {
+        service = factory.initializeService(accessToken);
+        try {
+            service.calendars().delete(calendarId).execute();
+        } catch (IOException e) {
+            throw new CalendarAccessException("Error deleting secondary calendar", e.getMessage());
+        }
+    }
+
+    @Override
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method length is 16 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)